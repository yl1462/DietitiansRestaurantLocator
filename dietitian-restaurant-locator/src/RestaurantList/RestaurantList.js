import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import config from '../config'

class RestaurantList extends Component {  
  handleDelete = (id) => {  
  fetch(`${config.API_ENDPOINT}/api/restaurant/${id}`, {
        method:'DELETE',
        headers:{
            'Content-Type':'application/json'
        }
    })
    
    .then(() => {
      this.props.deleteRestaurant(id)
      this.props.history.push('/home')
    })
    .catch(err => {
        this.setState({err})
    })
  }

  render() {
    return (
      <div className="restaurantList">

        {/* each restaurant shows the name*/}
        <ul>
          {
            this.props.restaurants.map((restaurant, index) => (
              <li key={index}>
                <p className="restaurantName">
                  {restaurant.the_restaurant}<br />
                </p>
                <p>
                  {/* let user edit or delete their restaurant items */}
                  {restaurant.type}<br />
                  <button><Link to={{ pathname: `/edit/${restaurant.id}`, state: { restaurant, index } }}>Edit</Link></button> <button onClick={() => this.handleDelete(restaurant.id)}>Delete</button>
                </p>
              </li>
            ))
          }
        </ul>
        {/* add new restaurant button on top of the page */}
        <button><Link to='/add'>Add New</Link></button>
      </div>

    );
  }
}

export default RestaurantList;
