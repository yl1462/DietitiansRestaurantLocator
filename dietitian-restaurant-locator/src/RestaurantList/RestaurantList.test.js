import React from 'react';
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RestaurantList from './RestaurantList'

describe('Component does not crash', () => {
  //test if page loads
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <BrowserRouter>
        <RestaurantList restaurants={[]}/>
      </BrowserRouter>,
      div
    )
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe(`RestaurantList component`, () => {
  //test if test data loads into each restaurant item
  const props = {
    restaurants: [
      {
        "name": "McDonald's",
        "type": "keto"
      },

      {
        "name": "Wendy's",
        "type": "mediterranean"        
      },

      {
        "name": "Five Guys",
        "type": "plant-based"        
      },
    ],
    history: { push: () => {} },
    match: { params: "" }
  }

  it('renders a restaurant in ul for each restaurants in array', () => {
    const ul = shallow(<RestaurantList {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
