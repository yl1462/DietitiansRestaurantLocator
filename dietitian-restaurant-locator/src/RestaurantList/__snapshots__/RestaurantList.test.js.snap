// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantList component renders a restaurant in ul for each restaurants in array 1`] = `
<ul>
  <li
    key="0"
  >
    <p
      className="restaurantName"
    >
      <br />
    </p>
    <p>
      keto
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 0,
                "restaurant": Object {
                  "name": "McDonald's",
                  "type": "keto",
                },
              },
            }
          }
        >
          Edit
        </Link>
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
  <li
    key="1"
  >
    <p
      className="restaurantName"
    >
      <br />
    </p>
    <p>
      mediterranean
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 1,
                "restaurant": Object {
                  "name": "Wendy's",
                  "type": "mediterranean",
                },
              },
            }
          }
        >
          Edit
        </Link>
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
  <li
    key="2"
  >
    <p
      className="restaurantName"
    >
      <br />
    </p>
    <p>
      plant-based
      <br />
      <button>
        <Link
          to={
            Object {
              "pathname": "/edit/undefined",
              "state": Object {
                "index": 2,
                "restaurant": Object {
                  "name": "Five Guys",
                  "type": "plant-based",
                },
              },
            }
          }
        >
          Edit
        </Link>
      </button>
       
      <button
        onClick={[Function]}
      >
        Delete
      </button>
    </p>
  </li>
</ul>
`;
